src/scope.rs

```diff
@@ -52,33 +84,169 @@ impl Scope {
     }
 
     pub fn get_let(&self, name: &str) -> Option<Gc<Expr>> {
-        match self {
-            Scope::None | Scope::Root(_) => Some(Gc::new(Expr {
-                range: None,
-                value: GcCell::new(None),
-                source: ExprSource::Literal {
-                    // TODO: add more keys here, such as `builtins`
-                    value: match name {
-                        "true" => NixValue::Bool(true),
-                        "false" => NixValue::Bool(false),
-                        "null" => NixValue::Null,
-                        _ => return None,
+        match self.get_definition(name) {
+            Definition::Expression(x) => Some(x),
+            _ => None,
+        }
+    }
+
+    /// how/if this name is defined
+    pub fn get_definition(&self, name: &str) -> Definition {
+        fn get_definition_rec(scope: &Scope, name: &str, possibly_dynamic: bool) -> Definition {
+            match scope {
+                Scope::None | Scope::Root(_) => Definition::Expression(Gc::new(Expr {
+                    range: None,
+                    value: GcCell::new(None),
+                    source: ExprSource::Literal {
+                        // TODO: add more keys here, such as `builtins`
+                        value: match name {
+                            "true" => NixValue::Bool(true),
+                            "false" => NixValue::Bool(false),
+                            "null" => NixValue::Null,
+                            // list obtained with `nix repl` tab completion
+                            "abort"
```

Comment:
Where do you recommend that we actually define these values?

Reply:
I don't know. Maybe we can ignore everything that starts with `__`, accept the builtins and hardcode the 3 or 4 remaining cases like true and false ?